### ISSUE TITLE
Desyncing

### ISSUE THREAD

**Issue by jaenster** [kolton/d2bot-with-kolbot#472]
In the file common\misc.js there is a function:
```
var Packet = { 
(....)
        flash: function (gid) {
		sendPacket(1, 0x4b, 4, 0, 4, gid);
	},
(....)
},
```
However due to the silent patches of blizzard we are desyncing like crazy. We could solve this in the past with flashing. However, that doesn't work anymore. With the current desyncing issue @ bnet, i think _Packet.flash_ in combination with _me.gid_ is not sufficient any more.

It's hard to know if you are de-synced since later the char suddenly warps to another location. A location where battle.net thinks the char is. So its hard to detect if the char is de-synced. However, in attacking a monster its quite easy to detect, if we don't inflict damage, we should be de-synced. Matter of fact, this check is already in the kill, clear and clearList function. It works perfect for the Hammerdin, but for melee attacks the good old re-sync packet isn't cutting it anymore

_Attack.js_
- _Packet.flash(me.gid);_ Line 187, function kill
- _Packet.flash(me.gid);_ Line 420, function clear
- _Packet.flash(me.gid);_ Line 574, function clearList

Example:

A amazon is attacking with charged strike, but bnet thinks im not next close enough to a monster (Baal for example), the client however does think its close and therefore does the bot. The bot detects de-syncing and a Hammerdin moves around however for other close range attacks it just sends the old re-sync packet, which isn't valid any more. 

What happens now is that Baal teleports away, the Amazon follows and she is re-synced by the movement and she inflict damage once again. However there are enough monsters that don't teleport or move.

There might be a solution for the attacks, to move like the Hammerdin randomly a bit and resume to attack. The bot will see our range is to big, and move closer to the monster, if necessary. However, by our movement we are re-synced @ bnet and we can hit the monster and do damage.

Long story short: Lets do the same for melee(=close range) attacks as we do for Hammerdins, move around a bit.

**Comment by jaenster** [comment]
``// Tele in random direction with close range attacks if we are desynced
                                if (gidAttack[i].attacks > 0 && gidAttack[i].attacks % ((target.spectype & 0x7) ? 4 : 2) === 0
                                    && ((Skill.getRange(Config.AttackSkill[(target.spectype & 0x7) ? 1 : 3]) < 4) // is it a close range attack?
                                    || (Config.AttackSkill[(target.spectype & 0x7) ? 1 : 3]) === 112)) { // or hammerdin?
                                    Pather.moveTo(me.x + rand(-1, 1) * 5, me.y + rand(-1, 1) * 5);
                                }``

That should solve the issue, for example for the _clear_ function @ attack
